test/check/check.go

```diff
@@ -125,6 +127,62 @@ func CheckOperatorIsReady(timeout time.Duration) {
 	}
 }
 
+func CheckForLeftoverObjects(currentVersion string) {
+	listOptions := client.ListOptions{}
+	key := opv1alpha1.SchemeGroupVersion.Group + "/version"
+	listOptions.SetLabelSelector(fmt.Sprintf("%s,%s != %s", key, key, currentVersion))
+
```

Comment:
You can iterate all this since the List() expectes a runtime.Object.

Reply:
test/check/check.go:133:17: cannot convert "github.com/kubevirt/cluster-network-addons-operator/vendor/k8s.io/api/apps/v1".DeploymentList literal (type "github.com/kubevirt/cluster-network-addons-operator/vendor
/k8s.io/api/apps/v1".DeploymentList) to type "github.com/kubevirt/cluster-network-addons-operator/vendor/k8s.io/apimachinery/pkg/runtime".Object:                                                                 
        "github.com/kubevirt/cluster-network-addons-operator/vendor/k8s.io/api/apps/v1".DeploymentList does not implement "github.com/kubevirt/cluster-network-addons-operator/vendor/k8s.io/apimachinery/pkg/runti
me".Object (DeepCopyObject method has pointer receiver)

Reply:
Yeah, but runtime.Object doesn't implement .Items()

Reply:
Is not that, since you are passing and interface you don't need to pass pointer, meaning just remove & from argument.